test.js

```diff
@@ -40,3 +40,14 @@ test('concurrency: 4', async t => {
 test('handles empty iterable', async t => {
 	t.deepEqual(await m([], mapper), []);
 });
+
+test('async with concurrency: 2', async t => {
+	const myInput = [100, 200, 10, 36, 13, 45];
+	const myMapper = value => {
+		return new Promise(resolve => {
+			setTimeout(() => resolve(value), value);
+		});
```

Comment:
Hi guys. FYI the original test in my app was failing about 20% of the time because of the elapse time for each promise was slightly varying between execution (promise was executing real code, not just delay). Just to say that a test with randomness may not always prove there is no issue. So I suggest to have two tests instead of only one: one with randomness and another one with hardcoded value as a regression.